Add docs for plugins with enterprise scheduled root rotation feature (#29608)
hashicorp/vault: website/content/docs/auth/azure.mdx

       https://127.0.0.1:8200/v1/auth/azure/role/dev-role
   ```

## Root credential rotation

If the mount is configured with credentials directly, the credential's key may be
rotated to a Vault-generated value that is not accessible by the operator.
This will ensure that only Vault is able to access the "root" user that Vault uses to
manipulate dynamic & static credentials.

```shell-session
vault write -f auth/azure/rotate-root
```

### Schedule-based root credential rotation

<EnterpriseAlert product="vault" />

The Azure auth engine supports configuring schedule-based automatic
credential rotation for the root credential with the
[rotation_schedule](/vault/api-docs/auth/azure#rotation_schedule) field.
For example:

```shell-session
$ vault write auth/azure/config \

>     rotation_schedule="0 * * * SAT"

Reviewer: Should we make it more obvious that this is not a complete example of configuring the mount? Something like ellipses maybe?

```suggestion
    rotation_schedule="0 * * * SAT"
    # ...
```

Author: We can, although this is also acceptable as a configuration to send, assuming you've already configured the backend.

Author: i added ellipses to the example requests to indicate that more fields can/should exist if needed/appropriate

       https://127.0.0.1:8200/v1/auth/azure/role/dev-role
   ```

## Root credential rotation

> If the mount is configured with credentials directly, the credential's key may be
> rotated to a Vault-generated value that is not accessible by the operator.
> This will ensure that only Vault is able to access the "root" user that Vault uses to
> manipulate dynamic & static credentials.

Reviewer 2: ```suggestion
Mounts can rotate root credential keys configured directly within the mount.
Rotating to a Vault-generated key makes the key value inaccessible to the
operator and ensures only Vault can operate as a root user to manipulate dynamic
and static credentials.
```
Style correction: write in active voice, avoid possessives

Author: applied to auth/azure and secrets/azure

```shell-session
vault write -f auth/azure/rotate-root
```

### Schedule-based root credential rotation

> <EnterpriseAlert product="vault" />

Reviewer 2: We have dedicated partials for enterprise alerts depending on whether the feature is available in HVD **and** enterprise or enterprise only: `@include 'alerts/enterprise-only.mdx'` or `@include 'alerts/enterprise-and-hcp.mdx`

Author: applied in 5 places across the 4 plugins (twice in auth/ldap)

> The Azure auth engine supports configuring schedule-based automatic
> credential rotation for the root credential with the
> [rotation_schedule](/vault/api-docs/auth/azure#rotation_schedule) field.
> For example:

Reviewer 2: ```suggestion
Use the [`rotation_schedule`](/vault/api-docs/auth/azure#rotation_schedule) field
to configure schedule-based, automatic credential rotation for root credentials in
the Azure auth engine. For example, the following command set the rotation to
occur every Saturday at midnight (00:00):
```
Style correction: write in active voice

Author: applied (with necessary plugin name and path edits) to all four plugins

```shell-session
$ vault write auth/azure/config \
    ...
    rotation_schedule="0 * * * SAT"
    ...
```

> This configuration will set the role's credential rotation to occur on Saturday
> at 00:00.

Reviewer 2: ```suggestion
```

Author: as above

Additionally, this schedule-based approach allows for optionally configuring a
[rotation_window](/vault/api-docs/auth/azure#rotation_window) in which
the automatic rotation is allowed to occur. For example:

```shell-session
$ vault write auth/azure/config \
    ...
    rotation_window="1h" \
    rotation_schedule="0 * * * SAT"
    ...
```

> This configuration will set rotations to occur on Saturday at 00:00. The 1
> hour `rotation_window` will prevent the rotation from occuring after 01:00. If
> the static role's credential is not rotated during this window, due to a failure
> or otherwise, it will not be rotated until the next scheduled rotation.

Reviewer 2: ```suggestion
```suggestion
Scheduled root credential rotation can also set a
[rotation_window](/vault/api-docs/auth/azure#rotation_window) during which the
scheduled rotation is allowed to occur. Vault will stop trying to rotate the
credential once the window expires. For example, the following command tells
Vault to rotate the credential on Saturday at midnight, but only within the span
of an hour. If Vault cannot rotate the credential by 1:00, due to a failure
or otherwise, Vault will stop trying to rotate the credential until the next
scheduled rotation.
```

Style correction: write in active voice, avoid possessives, explain code examples before presenting them
```

Author: changed in all four plugins

> !> The `rotation_period` and `rotation_schedule` fields are
> mutually exclusive. One of them must be set but not both.

Reviewer 2: ```suggestion
```
Nothing in the text above references `rotation_period`. Someone scanning quickly may mistake `rotation_period` for `rotation_window` and become confused.

> Root rotation can be temporarily disabled by setting [disable_automated_rotation](/vault/api-docs/auth/azure#disable_automated_rotation)
> to `true`. This will prevent any rotation of the root credential until the field is set to `false`. Note: This will reset
> the TTL if `rotation_period` is being used.

Reviewer 2: ```suggestion
You can temporarily disable root rotation by setting
[`disable_automated_rotation`](/vault/api-docs/auth/azure#disable_automated_rotation)
to `true`. Setting the `disable_automated_rotation` field prevent any rotation
of the root credential until the field is reset to `false`. If you use
`rotation_period`, setting `disable_automated_rotation` also resets the credential
TTL.
```
Style correction: write in active voice, avoid "this" as a pronoun

> For more details on this operation, please see the
> [Root Credential Rotation](/vault/api-docs/auth/azure#rotate-root) API docs.

Reviewer 2: ```suggestion
For more details on rotating root credentials in the Azure plugin, refer to the
[Root credential rotation](/vault/api-docs/auth/azure#rotate-root) API docs.
```
Style correction: avoid "this" as a pronoun
